File: 2024/day03/src/part1.rs
use std::fs;
use regex::Regex;

fn main() {
    //let data = fs::read_to_string("example.txt").unwrap();
    let data = fs::read_to_string("input.txt").unwrap();
    //println!("data: {}", data);

    let re = Regex::new(r"mul\((\d+),(\d+)\)").unwrap();

    let result: usize = re.captures_iter(&data)
                                .map(|capture| 
                                    capture[1].parse::<usize>().unwrap() * 
                                    capture[2].parse::<usize>().unwrap())
                                .sum();

    

    println!("result: {}", result);
    // result: 174336360
}

File: day01/part1.py
# data = open("example.txt").read().strip()
data = open("input.txt").read().strip()

data = [int(line) for line in data.splitlines()]
print(data)

result = 0
prev = None
for curr in data:
    if prev and curr > prev:
        result += 1
    prev = curr

print(result)
